src/main/java/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -2273,6 +2273,26 @@ public static <T> T[] eachWithIndex(T[] self, @ClosureParams(value=FromString.cl
         return self;
     }
 
+    /**
+     * Iterates through an int[],
+     * passing each int and the element's index (a counter starting at
+     * zero) to the given closure.
+     * <pre class="groovyTestCase">
+     * int result = 0
+     * [0, 1, 2].eachWithIndex{ x, index {@code ->} result += x * index }
+     * assert result == 5
+     * </pre>
+     *
+     * @param self    an int array
+     * @param closure a Closure to operate on each item
+     * @return the self Object
+     * @since 1.0
+     */
+    public static int[] eachWithIndex(int[] self, @ClosureParams(value=FromString.class, options="int,Integer") Closure closure) {
```

Comment:
I think "int" should be "Integer".  The closure params are passed as objects, so there is no optimization like `Consumer` versus `IntConsumer`.